client/call_graph.cpp

```diff
@@ -0,0 +1,107 @@
+#include "call_graph.hpp"
+
+#include <algorithm>
+
+namespace HawkTracer
+{
+namespace client
+{
+
+std::vector<std::pair<std::shared_ptr<CallGraph::TreeNode>, int>> 
+    CallGraph::make(std::vector<NodeData>& events)
+{
+    std::sort(events.begin(), events.end(), 
+            [](const NodeData& e1, const NodeData& e2){
+                return e1.start_ts < e2.start_ts;
+            });
+    for (const auto& event : events)
+    {
+        _add_event(event);
+    }
+
+    return _root_calls;
+}
+
+void CallGraph::_add_new_calltree(NodeData& node_data)
+{
+    std::string label = node_data.label;
+    auto call = std::find_if(_root_calls.begin(), _root_calls.end(),
```

Comment:
this `find_if` is very similar to the one in `_add_new_event_call`. Could it be moved to a new function?

Reply:
Actually, the whole function is similar, maybe we could move `if..else` as well.